Python/sysmodule.c

```diff
@@ -96,6 +96,80 @@ PySys_SetObject(const char *name, PyObject *v)
         return PyDict_SetItemString(sd, name, v);
 }
 
+static PyObject *
+sys_breakpointhook(PyObject *self, PyObject *args, PyObject *keywords)
+{
+    char *envar = getenv("PYTHONBREAKPOINT");
```

Comment:
Hum, you should use Py_GETENV() since PYTHON* env vars should be ignored when using python3 -E.

Reply:
I wonder if `PYTHONBREAKPOINT` should be ignored with -E.  I suppose it should, but a use case would be a production system that wants a safe environment but *also* wants to ensure that no stray `breakpoint()` calls stop the process.  That was @ncoghlan 's main use case for `PYTHONBREAKPOINT=0`.

Maybe `-E` should effectively set `PYTHONBREAKPOINT=0`?

Reply:
My suggestion:

- use Py_GETENV, so -E ignores `PYTHONBREAKPOINT`, but still defaults to dropping in to pdb
- have -I (isolated mode) imply PYTHONBREAKPOINT=0 with no way to override it at runtime

That would probably require a field in the CoreConfig struct to say whether or not to even consider checking PYTHONBREAKPOINT for breakpoints or not: https://github.com/python/cpython/blob/master/Include/pystate.h#L27

(Either way, this should probably go back to python-dev for discussion & resolution, since it isn't clear to me what the right answer actually is)